annnufan/assignment1/solution_2.cpp

```diff
@@ -0,0 +1,65 @@
+#include <string>
+#include <iostream>
+
+template <typename T>
+class node;
+
+
+template <typename T>
+class node {
+	T value;
+	node<T>* next;
+public:
+	node(T x, node<T>* nxt = nullptr) : value(x), next(nxt) {}
+	T get_value() {
+		return value;
+	}
+
+	node<T>* get_next() {
+		return next;
+	}
+
+	void set_value(T x) {
+		value = x;
+	}
+
+	void set_next(node<T>* nxt) {
+		next = nxt;
+	}
+
+};
+
+template <typename T>
+node<T>* find_kth_to_last(node<T>* head, int k) {
+	node<T>* tail = head;
+	node<T>* kth = head;
+	int between = 0;

```

Comment:
`tail` and `kth` are good names, but the name `between` does not help understand the purpose of this variable.
Maybe `count` is better?